src/pybind/mgr/dashboard/frontend/src/app/ceph/cluster/upgrade/upgrade.component.html

```diff
@@ -0,0 +1,101 @@
+<ng-container *ngIf="{upgradeInfo: upgradeInfo$ | async, error: upgradeInfoError$ } as upgrade">
+  <div class="row h-25 ms-1"
+       *ngIf="!upgrade.upgradeInfoError && upgrade.upgradeInfo as upgradeInfo; else checkUpgrade">
+    <ng-container *ngIf="healthData$ | async as healthData">
+      <div class="col-lg-3 h-50 d-flex flex-column border justify-content-center align-items-center">
+        <span class="bold">Current Version</span>
+        <span class="mt-1">{{ version }}</span>
+      </div>
+      <div class="col-lg-3 h-50 d-flex flex-column border justify-content-center align-items-center">
+        <span class="bold">Cluster Status</span>
+        <ng-template #healthChecks>
+          <ul>
+            <li *ngFor="let check of healthData.health.checks">
+              <span [ngStyle]="check.severity | healthColor"
+                    [class.health-warn-description]="check.severity === 'HEALTH_WARN'">
+              {{ check.type }}</span>: {{ check.summary.message }}
+            </li>
+          </ul>
+        </ng-template>
+        <div class="info-card-content-clickable mt-1"
+             [ngStyle]="healthData.health.status | healthColor"
+             [ngbPopover]="healthChecks"
+             popoverClass="info-card-popover-cluster-status">
+             {{ healthData.health.status | healthLabel | uppercase }}
+          <i *ngIf="healthData.health?.status !== 'HEALTH_OK'"
+             class="fa fa-exclamation-triangle"></i>
+        </div>
+      </div>
+      <div class="col-lg-3 h-50 d-flex flex-column border justify-content-center align-items-center">
+        <span class="bold">Upgrade Status</span>
+        <ng-container *ngIf="upgradeInfo.versions.length > 0; else noUpgradesAvailable">
+          <span class="mt-2"
+                i18n>
+          <i [ngClass]="[icons.up]"
+             class="text-info"></i>
+            Upgrade available</span>
+          <div i18n-ngbTooltip
+               [ngbTooltip]="(healthData.mgr_map | mgrSummary).total <= 1 ? 'To upgrade, you need minimum 2 mgr daemons.' : ''">
+            <button class="btn btn-accent mt-2"
+                    id="upgrade"
+                    aria-label="Upgrade now"
+                    [disabled]="(healthData.mgr_map | mgrSummary).total <= 1"
```

Comment:
This query seems costly just to get count of the mgr, isn't it? Maybe we can get it from orch api, wdyt?

Reply:
we are also displaying the health status and to get the health status i'll anyway need to get the healthData. So its less costlier than adding another api call.